pykokkos/lib/ufuncs.py

```diff
@@ -0,0 +1,61 @@
+import pykokkos as pk
+
+# NOTE: it might be slick if we could encapsulate
+# the workload code inside of the ufunc definitions,
+# but we currently require global scoping
+
+
+@pk.workload
+class ReciprocalView1D:
+    def __init__(self, view: pk.View1D[pk.double]):
+        # TODO: why doesn't view have a "size"
+        # attribute? it is basically the shape product
+        # in higher dims
+        self.threads: int = view.shape[0] # type: ignore
+        self.view: pk.View1D[pk.double] = view
+
+    @pk.main
+    def run(self) -> None:
+        pk.parallel_for(self.threads, self.pfor)
+
+    @pk.workunit
+    def pfor(self, tid: int) -> None:
+        self.view[tid] = 1 / self.view[tid] # type: ignore
+
+
+# TODO: how are we going to "ufunc dispatch" when the view
+# has a different type/precision? i.e., something other than
+# pk.double?
```

Comment:
what would you think about using Python single dispatch here: https://docs.python.org/3/library/functools.html#functools.singledispatch ?

Reply:
I'm not too familiar with this. I looked through the docs quickly, it seems that it will look at the type annotations and call the right function based on the type at runtime? If so that seems good, I think we should use it.

Reply:
Of course, we'd still need to implement a different kernel for each view type